exchanges/stream/websocket.go

```diff
@@ -545,59 +517,49 @@ func (w *Websocket) FlushChannels() error {
 // 1 slot buffer means that connection will only write to trafficAlert once per trafficCheckInterval to avoid read/write flood in high traffic
 // Otherwise we Shutdown the connection after trafficTimeout, unless it's connecting. connectionMonitor is responsible for Connecting again
 func (w *Websocket) trafficMonitor() {
-	if w.IsTrafficMonitorRunning() {
-		return
-	}
-	w.setTrafficMonitorRunning(true)
 	w.Wg.Add(1)
-
-	go func() {
-		t := time.NewTimer(w.trafficTimeout)
-		for {
+	t := time.NewTimer(w.trafficTimeout)
+	for {
+		select {
+		case <-w.ShutdownC:
+			if w.verbose {
+				log.Debugf(log.WebsocketMgr, "%v websocket: trafficMonitor shutdown message received", w.exchangeName)
+			}
+			t.Stop()
+			w.Wg.Done()
+			return
+		case <-time.After(trafficCheckInterval):
```

Comment:
This case seems to just bleed the traffic alert buffer by 1 and continues, this seems to already be handled below by the timer, could probably remove this case altogether?

Reply:
Need to add a comment to explain why it's here.
It's about the cadence.
If the timeout went off, and we reset the timer at that point, there's no relationship between when we got traffic and the *next* traffic timeout.
This case is to ensure that as close as possible we reset the timer when we get traffic to avoid giving double timeouts.
e.g. If we removed this check:
10 second timeout
Traffic arrives in first 1 second and not again after
9 seconds later we'd reset for another 10 seconds

Reply:
Gotcha thanks!